modules/core/src/smithy4s/Hints.scala

```diff
@@ -25,11 +25,23 @@ package smithy4s
   * in a label, a regex pattern some string should abide by, a range, etc)
   *
   * This `Hints` interface is a container for hints.
+  *
+  * Under the hood, the hints are composed of two maps : one for member-level hints,
+  * one for target-level hints.
   */
 trait Hints {
   def isEmpty: Boolean
   def all: Iterable[Hints.Binding]
+
+  def memberHintsMap: Map[ShapeId, Hints.Binding]
+  def targetHintsMap: Map[ShapeId, Hints.Binding]
```

Comment:
the distinction between member hints and target hints is here. What we've learnt is that most of the time, the distinction won't matter in interpreters. However, it may matter (in cases such as Xml, where `XmlName` MUST be handled differently based on whether it's on a shape or a member). 

Allowing for the `Hints` construct to own this distinction minimises the impact to existing SchemaVisitors, as members can not only be field members, but also list members, map key/value, etc. Handling all these cases just to make Xml happy would induce a lot of frustrating changes.